Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+
+            if (SMcomponent.DamageArchived < SupermatterComponent.ExplosionPoint)
+            {
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-safe-allert"), "Supermatter", false);
+                SMcomponent.FinalCountdown = false;
+                return;
+            }
+            else if (RoundSeconds >= 5 && SMcomponent.SpeakAccumulator >= 5)
+            {
+                SMcomponent.SpeakAccumulator -= 5;
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-seconds-before-delam", ("Seconds", RoundSeconds)), "Supermatter", false);
+            }
+            else if (RoundSeconds <  5 && SMcomponent.SpeakAccumulator >= 1)
+            {
+                SMcomponent.SpeakAccumulator -= 1;
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-seconds-before-delam", ("Seconds", RoundSeconds)), "Supermatter", false);
+            }
+
+            if (SMcomponent.DelamTimerAccumulator >= SupermatterComponent.DelamTimerTimer)
+            {
+                if (_atmosphereSystem.GetTileMixture(Xform.Coordinates) is { } mixture)
+                {
+                    if (mixture.TotalMoles >= SupermatterComponent.MolePenaltyThreshold)
+                    {
+                        EntityManager.SpawnEntity("Singularity", Xform.Coordinates);
+                        return;
+                    }
+                }
+                //TODO: tune this after explosion refactor
+                _explosions.SpawnExplosion(Xform.Coordinates, 75, 75, 75, 100);
+                EntityManager.QueueDeleteEntity(Uid);
+            }
+
+            if (SMcomponent.FinalCountdown && !SMcomponent.AlarmPlaying && RoundSeconds <= 13)
+            {
+                SMcomponent.AlarmPlaying = true;
+                SoundSystem.Play(Filter.Pvs(Uid, 2), SMcomponent.DelamAlarm.GetSound(), Uid);
+            }
+        }
+
+        /// </summary>
+        /// Determines if an entity can be dusted
+        /// </summary>
+        public bool CannotDestroy(EntityUid Uid)
+        {
+            bool Tag = false;
+            bool Static = false;
+
+            if (EntityManager.HasComponent<TagComponent>(Uid))
+            {
+                Tag = EntityManager.GetComponent<TagComponent>(Uid).HasTag("SMimmune");
+            }
+
+            if (EntityManager.HasComponent<PhysicsComponent>(Uid))
+            {
+                Static = EntityManager.GetComponent<PhysicsComponent>(Uid).BodyType.ToString() == "Static";
+            }
+
+            return Tag || Static;
+        }
+
+        private void OnCollideEvent(EntityUid uid, SupermatterComponent supermatter, StartCollideEvent args)
+        {
+            EntityUid Target = args.OtherFixture.Body.Owner;
+            if (!supermatter.Whitelist.IsValid(Target) || CannotDestroy(Target) || Target.IsInContainer()) return;
```

Comment:
Why have a whitelist and check for the tag? That makes no sense.

Also bodies in containers shouldn't be collidable so you shouldn't need to check for it.